code/modules/research/designs.dm

```diff
@@ -628,6 +597,14 @@
 	build_path = /obj/item/ammo_magazine/pulse
 	sort_string = "TBECD"
 
+/datum/design/item/weapon/pulsehighvel
+	name = "rifle high-velocity magazine"
+	id = "pulsehighvel"
+	req_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 4)
+	materials = list(MATERIAL_PLASTEEL = 15000, MATERIAL_DIAMOND = 5000, MATERIAL_SILVER = 1000) // Same thing here. It might need tweaking with exotic materials to stop people getting a lot of it early on though.
```

Comment:
this might be excessive, but we'll see how it goes. 

change the name to high velocity pulse magazine though